### PR TITLE
Add option to create anonymous exams

### PR DESCRIPTION
**Pull request by sunye**
PR summary:

- new option to allow creating anonymous exams
- move changelog to a specific file
- split the mixed French and English example in two, one for each language

Anonymous exams cannot have name nor id fields, only an anonymity number field. When the `anonymous` option is used, the result is the following:

anonymat

### CODE REVIEW

File: exam-example-en.tex [hunk 1]
```diff
@@ -1,7 +1,10 @@
 % Options:
-% withid: adds a header line for writing the student's ID (if a value is given, will limit the amount of boxes)
-% We recommend to keep the twoside option as it permits to add a specific header (student' name) on each new page.
-\documentclass[a4paper,twoside,withid=8]{exam}
+% withid: adds a header line for writing the student's ID (if a value is given, will limit the number of boxes)
+% anonymous: replaces student name fieds by an anonimity number (if a value is given, it specifies the number of boxes)
+%
+% We recommend to keep the twoside option as it allows to add a specific header (student' name) on each new page.
+\documentclass[a4paper,twoside,anonymous]{exam}
```
Comment (arnobl) [inline exam-example-en.tex]: Can you put this line as a comment and keep the previous `\documentclass[a4paper,twoside,withid=8]{exam}` as the default parameters? According to usages, most of the exams are not anon.

File: exam.cls [hunk 2]
```diff
@@ -120,9 +100,13 @@
 {%
   \setlength{\tabcolsep}{0.2em}%
   \begin{tabular}{rl}
-    \textbf{\small NOM :} & \zsavepos{lastnamebox}\multido{}{\nbBoxesHeaderLine}{~\boxStudentID}\\
-    \textbf{\small PRÉNOM :} & \zsavepos{firstnamebox}\multido{}{\nbBoxesHeaderLine}{~\boxStudentID}\\
-    \ifheaderWithID \textbf{\small NUM ETD :} & \zsavepos{stdidbox}\multido{}{\nbBoxesStudentID}{~\boxStudentID} \else \\  \fi
+    \ifanonymousExamen%
+      \textbf{\small \No ANONYMAT~:} & \zsavepos{stdidbox}\multido{}{\nbBoxesAnonimityNumber}{~\boxStudentID}\\
+    \else
+      \textbf{\small NOM :} & \zsavepos{lastnamebox}\multido{}{\nbBoxesHeaderLine}{~\boxStudentID}\\
```
Comment (arnobl) [inline exam.cls:106]: Does it work with the english template?

**Comment by arnobl** [comment]
Thank you for this work! I wrote two comments.

File: exam.cls [hunk 3]
```diff
@@ -34,6 +10,10 @@
 \newif\ifheaderWithID
 \newcommand{\nbBoxesStudentID}{}
 
+\newif\ifanonymousExamen
+\newcommand{\nbBoxesAnonimityNumber}{}
+
+\DeclareOptionX{anonymous}[4]{\anonymousExamentrue\renewcommand{\nbBoxesAnonimityNumber}{#1}}
```
Comment (arnobl) [inline exam.cls:16]: Maybe the number of boxes  should be a parameter (with a default value)?

File: exam.cls (hunk 2 above)
Comment (arnobl) [reply exam.cls:106]: But may be part of another PR: add another parameter to define the language of the template

File: exam-example-en.tex (hunk 1 above)
Comment (sunye) [reply exam-example-en.tex]: Oops, my bad, I forgot to change the commented line

File: exam.cls (hunk 3 above)
Comment (sunye) [reply exam.cls:16]: I really don't like the option name, but I couldn't find a better identifier. Maybe using two options is a better choice

Comment (sunye) [reply exam.cls:16]: Maybe a better choice would be to allow adding fields independently of the class options.

File: exam.cls (hunk 2 above)
Comment (sunye) [reply exam.cls:106]: Is there an English template?

**Comment by arnobl** [comment]
Sorry I did some changes that led to conflicts.
Can you merge the conflicts? (but before pull the changes I made on this PR)
Just 
```
%\documentclass[a4paper,twoside,anonymous]{correctexam}
\documentclass[a4paper,twoside,withid=8]{correctexam}
```
in the example file I think.

**Comment by sunye** [comment]
I think it's fine now

**Review by arnobl — APPROVED** [review]

**Comment by arnobl** [comment]
Thank you!

**Pull request merged**